src/apps/integrations/prolific/service/prolific.py

```diff
@@ -0,0 +1,79 @@
+import ast
+from http.client import HTTPException
+import uuid
+
+import requests
+from fastapi import HTTPException
+
+from apps.applets.service.applet import AppletService
+from apps.integrations.crud.integrations import IntegrationsCRUD
+from apps.integrations.db.schemas import IntegrationsSchema
+from apps.integrations.domain import AvailableIntegrations
+from apps.integrations.prolific.domain import ProlificCompletionCode, ProlificCompletionCodeList, ProlificIntegration, PublicProlificIntegration
+from apps.users.domain import User
+
+
+class ProlificIntegrationService:
+    def __init__(self, applet_id: uuid.UUID, session) -> None:
+        self.applet_id = applet_id
+        self.session = session
+        self.type = AvailableIntegrations.PROLIFIC
+
+    async def create_prolific_integration(self, api_key: str) -> ProlificIntegration:
+        prolific_response = requests.get(
+            "https://api.prolific.com/api/v1/users/me/",
+            headers={"Authorization": f"Token {api_key}", "Content-Type": "application/json"},
+        )
+
+        if prolific_response.status_code != 200:
+            raise HTTPException(status_code=prolific_response.status_code, detail="Prolific token is invalid")
+
+        integration_schema = await IntegrationsCRUD(self.session).create(
+            IntegrationsSchema(
+                applet_id=self.applet_id,
+                type=self.type,
+                configuration={
```

Comment:
create ProlificIntegrationConfig pydantic model, serialize as json, then after retrieve unserialize json and parse using pydantic model